### PR TITLE
Nix test basic

### PR DESCRIPTION
**Pull request by gila**

### CODE REVIEW

File: nix/test/basic/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+TODO:
+[ ] We should write some high-level python code that we can reuse in the tests.
+[ ] We need to figure out how many tests we want to run within a single run
+[ ] We will need more gRPC methods to build more sophisticated tests
+[ ] A test where we set up and deploy, k8s first, and then install mayastor would be very nice but for sure should be done last.
+
+
+
+Note, this will only work after a "cargo build --all" as it the
```
Comment (blaisedias) [inline nix/test/basic/README.md:23]: missing word 'builds'

File: nix/test/basic/default.nix [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#
+# TODO: not sure if we need to import the sources
+#
+let
+  sources = import ./../../../nix/sources.nix;
```
Comment (blaisedias) [inline nix/test/basic/default.nix:5]: why leading './..' instead of just '../'

File: nix/test/basic/fio_nvme_basic.nix [hunk 3]
```diff
@@ -0,0 +1,109 @@
+{ pkgs, lib, ... }:
+let
+  targetIp = "192.168.0.1";
```
Comment (blaisedias) [inline nix/test/basic/fio_nvme_basic.nix:3]: Doesn't a fixed ip address limit where this test can be deployed?

File: nix/test/basic/fio_nvme_basic.nix [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  initiatorIp = "192.168.0.2";
```
Comment (blaisedias) [inline nix/test/basic/fio_nvme_basic.nix:4]: Doesn't a fixed ip address limit where this test can be deployed?

File: nix/test/basic/README.md (hunk 1 above)
Comment (gila) [reply nix/test/basic/README.md:23]: fixed, thanks.

File: nix/test/basic/default.nix (hunk 2 above)
Comment (gila) [reply nix/test/basic/default.nix:5]: that is nix-ism. All imports that refer to a file always start with "./" regardless of the path itself.

File: nix/test/basic/fio_nvme_basic.nix (hunk 3 above)
Comment (gila) [reply nix/test/basic/fio_nvme_basic.nix:3]: No, the network is a VLAN -- and is isolated from the host. You can not reach this network with your own host/machine.

File: nix/test/basic/fio_nvme_basic.nix (hunk 4 above)
Comment (gila) [reply nix/test/basic/fio_nvme_basic.nix:4]: Ideally, we don't have fixed IPs however, mayastor needs to have an IP in order to run and we don't have code that can "auto-discover" the IPs (as they are passed in as ENV through k8s).

**Pull request merged**